src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -65,14 +68,91 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsStandbyConfiguration = true;
             }
 
-            options.IsFileSystemReadOnly = IsZipDeployment();
+            options.IsFileSystemReadOnly = IsZipDeployment(options);
         }
 
-        private bool IsZipDeployment()
+        private bool IsZipDeployment(ScriptApplicationHostOptions options)
         {
-            return Utility.IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
+            // Check app settings for run from package.
+            bool runFromPkgConfigured = Utility.IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
                 Utility.IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteAltZipDeployment)) ||
                 Utility.IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteRunFromPackage));
+
+            if (!_environment.IsLinuxConsumption())
+            {
+                // This check is strong enough for SKUs other than Linux Consumption.
+                return runFromPkgConfigured;
+            }
+
+            // The following logic only applies to Linux Consumption, since currently SCM_RUN_FROM_PACKAGE is always set even if we are not using it.
+            if (runFromPkgConfigured)
+            {
+                return true;
+            }
+
+            // If SCM_RUN_FROM_PACKAGE is set to a valid value and the blob exists, it's a zip deployment.
+            var url = _environment.GetEnvironmentVariable(ScmRunFromPackage);
+            if (string.IsNullOrEmpty(url))
+            {
+                LinuxContainerEventGenerator.LogInfo($"{nameof(ScmRunFromPackage)} is empty.");
+            }
+            else
+            {
+                LinuxContainerEventGenerator.LogInfo($"{nameof(ScmRunFromPackage)} = {url.Substring(0, 25)}");
+            }
```

Comment:
probably dont need this explicit log since if url was empty that would have been logged already